packages/client/src/components/edit-user-data-dialog.vue

```diff
@@ -134,16 +135,12 @@ const formData = computed<
     delegate: {
       label: t("auth.nameOfDelegate"),
       infoLabel: t("auth.delegateLabel"),
-      rules:
-        newUserData.value.dateOfBirth &&
-        new Date().getUTCFullYear() -
-          new Date(newUserData.value.dateOfBirth).getUTCFullYear() <
-          18
-          ? [requiredRule]
-          : undefined,
+      rules: [requireIfUnderage(newUserData.value.dateOfBirth)],
```

Comment:
Ah wait, noticed a problem here: by doing this, the validation rule is basically taking in the value currently set inside the `newUserData.value.dateOfBirth`. The rule set here should be different:
```suggestion
      rules: requireIfUnderage(newUserData.value.dateOfBirth) ? [required] : undefined,
```
Because the way it is coded now it simply does a check on the date of birth if the user is > 18 year or not.
With no check if the delegate has been set.

Reply:
Notice that change along is not enough probably, because of what `requireIfUnderage` returns

Reply:
The implementation of `requireIfUnderage` needs updates, not the usage.

Also, we need to use the `reactive-rules` prop in order to immediately re-calculate the rules when the birth date gets changed.

Reply:
See https://github.com/ilmercatinodellibro/ilmercatinodellibro/pull/86#discussion_r1618868179